charts/seed-monitoring/charts/core/charts/prometheus/templates/config.yaml

```diff
@@ -366,10 +366,14 @@ data:
       - source_labels: [ __meta_kubernetes_pod_name ]
         action: keep
         regex: vpn-shoot-(.+)
-      - source_labels: [ __meta_kubernetes_pod_name ]
+      - source_labels: [ __meta_kubernetes_pod_container_name ]
+        regex: (vpn-shoot-.*|vpn-shoot)
+        replacement: $1
+        action: replace
+      - source_labels: [__meta_kubernetes_pod_name,__meta_kubernetes_pod_container_name]
         target_label: __param_target
-        regex: (.+)
-        replacement: https://kube-apiserver:443/api/v1/namespaces/kube-system/pods/${1}/log?tailLines=1
+        regex: (.+);(.+)
+        replacement: https://kube-apiserver:443/api/v1/namespaces/kube-system/pods/${1}/log?container=${2}&tailLines=1
```

Comment:
Ok, so together with @istvanballok we investigated what the PR is trying to achieve, semantically. This tunnel probe scrape job tests the VPN connection indirectly. It goes through the `blackbox-exporter`. The URL is passed as a parameter in the prometheus scrape request to the blackbox exporter (specifically it's `/probe` endpoint), which then tries to reach the kube-apiserver through the VPN tunnel. It does not matter which container name is supplied here. When prometheus does service discovery, it selects a container in the `__meta_kubernetes_pod_container_name` label. Prior to the HA approach, that can also just  be an init container, not specifically the `vpn-shoot` container. I have attached a screenshot of Prometheus' service discovery page 
![image](https://github.com/gardener/gardener/assets/9338170/31f10d4c-1731-4744-a37a-8b9734ee7bfe)
From my understanding, just the second part of this change should be sufficient, you don't need to replace the container name label (and I'm not sure that's a good idea, I think the idiomatic way is to use `__tmp` labels). This is because what matters is the ability of the blackbox-exporter to reach the kube-apiserver through the VPN tunnel, not the logs themselves.

Reply:
> It probes three times for one pod.

Yep, so that's unnecessary for our use case here. I don't have an HA shoot at the moment, but can you look for a shoot pod running in the `kube-system` namespace on an HA shoot that also only exists once in regular shoots and has a single container? Then you could use that for the blackbox-exporter probe. It's not the logs that we care about, it's about whether or not the blackbox-exporter can reach the kube-apiserver running in the shoot control plane via the VPN tunnel. So as long as the probe is successful (200 return code), we know that the kube api server is reachable and that's all that matters in this case.

Reply:
> It probes three times for one pod.

Maybe we could improve that by keeping only the discovered target for the `vpn-shoot-init` container. Regarding the pod, in the non HA case, the pod name looks like this: `vpn-shoot-74967bbd9c-fzsql` and has one init container and one regular container. In the HA case it is a statefulset with one init container and 2 regular containers. We could filter for the pod name (either -0 for the statefulset, or something - something for the deployment) and the container name.

I haven't tested this suggestion below. Could you please check something like this, in both the HA and non-HA case:

```
- source_labels: [ __meta_kubernetes_pod_name, __meta_kubernetes_pod_container_name ]
  action: keep
  regex: vpn-shoot-(0|.+-.+);vpn-shoot-init
- source_labels: [ __meta_kubernetes_pod_name,__meta_kubernetes_pod_container_name ]
  target_label: __param_target
  regex: (.+)
  replacement: https://kube-apiserver:443/api/v1/namespaces/kube-system/pods/${1}/log?container=${2}&tailLines=1
  action: replace
```

Reply:
@istvanballok I tested locally. Just a minor change for this
`regex: (.+);(.+)`
It works as expected, gets one pod `vpn-shoot-*` and target for the `vpn-shoot-init` container. 
For HA: 
```
probe_success{instance="https://kube-apiserver:443/api/v1/namespaces/kube-system/pods/vpn-shoot-0/log?container=vpn-shoot-init&tailLines=1", job="tunnel-probe-apiserver-proxy", type="seed"}
1
```
For non-HA node, 
```

probe_success{instance="https://kube-apiserver:443/api/v1/namespaces/kube-system/pods/vpn-shoot-5c4799f895-qhhnw/log?container=vpn-shoot-init&tailLines=1", job="tunnel-probe-apiserver-proxy", type="seed"} | 1

```

Reply:
> I tested locally. Just a minor change...

Thank you for the fix. I think it looks good!